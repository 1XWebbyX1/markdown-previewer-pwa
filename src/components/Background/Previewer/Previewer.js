import React from 'react'
import marked from "marked";
import Toolbar from '../Toolbar/Toolbar'



marked.setOptions({
  breaks: true,
});

// INSERTS target="_blank" INTO HREF TAGS
const renderer = new marked.Renderer();
renderer.link = function (href, title, text) {
  return `<a target="_blank" href="${href}">${text}` + '</a>';
}
renderer.code = function(code, language) {
  return '<pre><code class=language-' + language + '>' + code + '</code></pre>';
}



class Preview extends React.Component {
  render() {
    return (
        <div className="preview">
            <Toolbar id="preview-tool" text='Previewer' onClick={this.props.onClick}/>
            <div id="view" dangerouslySetInnerHTML={{__html: marked(
            this.props.markdown, { renderer: renderer })}} />
       </div>
    );
  }
}

export default Preview;
