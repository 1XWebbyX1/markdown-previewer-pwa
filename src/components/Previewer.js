import React from 'react'
import marked from "marked";
import asyncComponent from './asyncComponent/async'


const Toolbar = asyncComponent(() =>
        import('./Toolbar').then(module => module.default)
);


const renderer = new marked.Renderer();


class Preview extends React.Component {
  render() {
    return (
        <div class="preview">
            <Toolbar id="preview-tool" text='Previewer'/>
            <div id="view" dangerouslySetInnerHTML={{__html: marked(
            this.props.markdown, { renderer: renderer })}} />
       </div>
    );
  }
}

export default Preview;
