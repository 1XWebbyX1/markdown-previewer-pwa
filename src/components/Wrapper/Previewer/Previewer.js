import React from 'react'
import marked from "marked"

const renderer = new marked.Renderer();

// INSERTS target="_blank" INTO HREF TAGS
renderer.link = function (href, title, text) {
  return `<a target="_blank" href="${href}">${text}` + '</a>';
}
renderer.code = function(code, language) {
  return '<pre><code class=language-' + language + '>' + code + '</code></pre>';
}

marked.setOptions({
  breaks: true,
});


const Previewer = ({markdown}) => (
  <div className="textarea preview">
    <div id="view" dangerouslySetInnerHTML={{__html: marked(
            markdown, { renderer: renderer })}} />
   </div>
)


export default Previewer;
