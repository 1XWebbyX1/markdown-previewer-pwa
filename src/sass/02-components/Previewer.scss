.preview {
  word-wrap: break-word;
  line-height: 1.4;
  margin-left: 0;
  //customizing styles for markdown library----------

  blockquote {
  border-left: 3px solid #e60099;
  background:  white;
  color: #e60099;
  padding: 0.2px 15px;
  margin-left: 25px;
}

code {
  background: white;
  word-wrap: break-word!important;
  color: #e60099;
  padding: 1px 4px 2px 4px;
}

pre {
  background: white;
  padding: 5px 0 5px 5px;
}

h1 {
  border-bottom: 2px solid white;
}

h2 {
  border-bottom: 1px solid white;
}

table {
  border-collapse: collapse;
}

td, th {
  border: 2px solid #99004d;
  padding: 10px;
}

}
