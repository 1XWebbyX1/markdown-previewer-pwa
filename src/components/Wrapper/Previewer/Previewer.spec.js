// External Depedencies
import React from 'react';
import { mount, configure } from 'enzyme';
// Our Dependencies
import { expect } from '../utils/chai';
import Adapter from 'enzyme-adapter-react-16';
import renderer from "react-test-renderer";
// Our Component
import Previewer from './Previewer';


configure({ adapter: new Adapter() });

describe('Previewer', () => {

  it("renders correctly", () => {
     const output = renderer.create(
         <Previewer markdown="mockMarkdown"/>
   ).toJSON();
     expect(output).to.matchSnapshot(); //to check inner html is set to mockMarkdown
  });

  it('type of markdown is not null or undefined', () => {
    const app = mount(<Previewer markdown="mockMarkdown"/>);
    expect(typeof app.props().markdown).to.not.equal('undefined');
    expect(typeof app.props().markdown).to.not.equal('null');
  });

})
