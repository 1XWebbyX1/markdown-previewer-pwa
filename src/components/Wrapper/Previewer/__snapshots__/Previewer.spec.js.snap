// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Previewer renders correctly 1`] = `
<div
  className="textarea preview"
>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>mockMarkdown</p>
",
      }
    }
    id="view"
  />
</div>
`;
